andela/jubilee-backend
# #168133969 Implement supplier/host signup feature

#### What does this PR do?
Enable suppliers/host to be able to signup to use the platform
#### Description of Task to be completed?
- Create the database models, migrations and establish appropriate relationships between tables
- Create an endpoint for registering suppliers
- Validate all user inputs before registration
- Implement middleware for checking if the user already exists
- Ensure that an email gets fired upon successful signup
- Write unit tests for this feature
- Clean up codebase to follow team agreed standards
- Document the new endpoint on swagger
#### How should this be manually tested?
- Clone the repo and cd into the apps working directory
- Checkout to the `ft-supplier-host-signup-168133969` branch
- Create a .env file and specify the variables described in the .env.sample inside this file.
- Run `docker-compose up --build` to build the docker image and spin up all the necessary services
- Alternative to running the app on docker, you can run `npm run start:dev`
- Run `npm run migrate:reset && npm run migrate && npm run seed` to create tables in your database and seed the database tables with the necessary app configs.
- Launch Postman/Insomnia and make a `POST` request to `localhost:3000/api/auth/signup/supplier` route using the fields described in the swagger document of this project.
- Upon successful registration, a JSON response should be received with updated user parameters.
- To run the unit tests, run `npm run test`.
#### Any background context you want to provide?
This application is bootstrapped using docker as described above, however, the app can still be run locally without docker using the `npm run start:dev` script
#### What are the relevant pivotal tracker stories?
[#168133969](https://www.pivotaltracker.com/story/show/168133969)
#### Screenshots (if appropriate)
![Screen Shot 2019-08-29 at 6 26 19 PM](https://user-images.githubusercontent.com/33099347/63962330-91af6780-ca8a-11e9-8a43-021335448676.png)
